opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/AssetRecordResource.java

```diff
@@ -125,7 +129,7 @@ public Response updateAssetRecord(@PathParam("nodeCriteria") final String nodeCr
         }
         if (modified) {
             LOG.debug("updateAssetRecord: assetRecord {} updated", assetRecord);
-            m_nodeDao.saveOrUpdate(node);
+            m_assetRecordDao.update(assetRecord);
```

Comment:
Is it possible that the node did not have already have an asset record? In this case we would need a save instead of an update.